scripts/lib/CIME/case/case.py

```diff
@@ -530,6 +533,71 @@ def _find_primary_component(self):
         return primary_component
 
 
+    __mod_match_re__ = re.compile(r"([^%]*[^0-9%]+)")
```

Comment:
To always match a model name, peel off any `%` stuff plus any tailing digits (e.g., `CAM60`). Since this is done consistently, no special cases for ww3 or fv3 are required.

Reply:
Can you add a comment here illustrating why this regular expression looks like it does? In particular, I'm confused by the initial `[^%]*`. A few examples would help here.

Reply:
Sorry, I'm still confused about why you have `[^%]*` in this regex, as opposed to just `[^0-9%]+`. Can you provide an example where the `[^%]*` is needed? Sorry to be harping on this, but I find that it's very hard to modify regexes later without sufficient examples illustrating why they look the way they look.

Reply:
`fv3gfs`
Your version would only pick up `fv`. Using greedy nature of regex processing, my version only chops trailing digits. Unless I goofed?

Reply:
Ah, thanks, that's exactly the example I was looking for. So a comment that would help me would be something like:

This regex picks up characters up to the first '%', but leaving off any digits immediately before the '%'. So, for example, 'CAM60%FOO' will pick up 'CAM', and 'FV3GFS%BAR' will pick up 'FV3GFS'.

If that is correct, can you please add a comment along those lines?

Reply:
The current comment states "Drop any trailing digits" (I fixed my typo). Is that not explicit enough? I don't really want to write a regex tutorial in the comment block. Currently, I have added this at the end:
```
# Note: interstitial digits are included (e.g., in FV3GFS).
```
Sufficient?

Reply:
Yes, your suggested addition to the comment is sufficient - thanks.